src/support/azext_support/custom.py

```diff
@@ -6,174 +6,276 @@
-            for (region, payload) in zip(quota_change_regions, quota_change_payload):
-                quota_change_requests.append({"region": region, "payload": payload})
-        quotaBody["quota_change_requests"] = quota_change_requests
-        body["quota_ticket_details"] = quotaBody
-
-    if is_technical_ticket(service) and technical_resource is not None:
-        body["technical_ticket_details"] = {"resource_id": technical_resource}
-
-    logger.debug("Sending create request with below payload: ")
-    logger.debug(json.dumps(body, indent=4))
-
-    if partner_tenant_id is not None:
-        external_bearer_token = get_bearer_token(cmd, partner_tenant_id)
-        return client.begin_create(support_ticket_name=ticket_name, create_support_ticket_parameters=body,
-                                   headers={'x-ms-authorization-auxiliary': external_bearer_token})
-
-    return client.begin_create(support_ticket_name=ticket_name, create_support_ticket_parameters=body)
-
-
-def create_support_tickets_communications(cmd, client,
-                                          ticket_name=None,
-                                          communication_name=None,
-                                          communication_body=None,
-                                          communication_subject=None,
-                                          communication_sender=None):
-    logger.warning("Command group 'az support tickets communications' will be replaced by" +
-                   " 'az support in-subscription communication'" +
-                   " and 'az support no-subscription communication'")
-    body = {}
-    body["sender"] = communication_sender
-    body["subject"] = communication_subject
-    body["body"] = communication_body
-
-    return client.begin_create(support_ticket_name=ticket_name, communication_name=communication_name,
-                               create_communication_parameters=body)
+def set_configured_defaults(args):
+    for arg_name, arg in args:
+        if arg_name == "contact_language":
+            arg.completer = getLanguage()
+        if arg_name == "contact_timezone":
+            arg.completer = getTimeZone()
+    return args
+
+
+class TicketUpdate(_Update):
+
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        return set_configured_defaults(args)
+
+
+class TicketUpdateNoSubscription(_UpdateNoSubscription):
+
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        return set_configured_defaults(args)
+
+
+class TicketCreate(_CreateTicket):
+
+    def _cli_arguments_loader(self):
+        args = super()._cli_arguments_loader()
+        return set_configured_defaults(args)
+
+    def pre_operations(self):
+        from azext_support._validators import _validate_tickets_create
+        from azure.cli.core.aaz import AAZUndefined
+
+        super().pre_operations()
```

Comment:
rewrote pre_operation don't need and should not call its original method.
followings are the same